index.bs

```diff
@@ -381,38 +378,45 @@ Abstract Operations {#abstract-opertaions}
-to provide mocking information about the [=magnetic field=] in the X, Y and Z axis
-for the purposes of testing a user agent's implementation of {{Magnetometer}} and {{UncalibratedMagnetometer}} APIs.
 
-<h3 id="mock-magnetometer-type">Mock Sensor Type</h3>
+This section extends [[GENERIC-SENSOR#automation]] by providing [=Magnetometer=]-specific virtual sensor metadata.
 
-The {{Magnetometer}} class has an associated [=mock sensor type=] which is
-<a for="MockSensorType" enum-value>"magnetometer"</a>, its [=mock sensor reading values=]
-dictionary is defined as follows:
+Magnetometer automation {#magnetometer-automation}
+-----------------------
 
-<pre class="idl">
-  dictionary MagnetometerReadingValues {
-    required double? x;
-    required double? y;
-    required double? z;
-  };
-</pre>
+The [=per-type virtual sensor metadata=] [=map=] must have the following [=map/entry=]:
+: [=map/key=]
+:: "`magnetometer`"
+: [=map/value=]
+:: A [=virtual sensor metadata=] whose [=virtual sensor metadata/virtual sensor type=] is [=Magnetometer=] and [=reading parsing algorithm=] is [=parse XYZ reading=].
 
-The {{UncalibratedMagnetometer}} class has an associated [=mock sensor type=] which is
-<a for="MockSensorType" enum-value>"uncalibrated-magnetometer"</a>, its [=mock sensor reading values=]
-dictionary is defined as follows:
+Uncalibrated Magnetometer automation {#uncalibrated-magnetometer-automation}
+------------------------------------
 
-<pre class="idl">
-  dictionary UncalibratedMagnetometerReadingValues {
-    required double? x;
-    required double? y;
-    required double? z;
-    required double? xBias;
-    required double? yBias;
-    required double? zBias;
-  };
-</pre>
+The [=per-type virtual sensor metadata=] [=map=] must have the following [=map/entry=]:
+: [=map/key=]
+:: "`uncalibrated-magnetometer`"
+: [=map/value=]
+:: A [=virtual sensor metadata=] whose [=virtual sensor metadata/virtual sensor type=] is [UncalibratedMagnetometer](#uncalibrated-magnetometer-interface) and [=reading parsing algorithm=] is [=uncalibrated magnetometer reading parsing algorithm=].
+
+<h4 dfn>Uncalibrated Magnetometer reading parsing algorithm</h4>
+<div algorithm="uncalibrated magnetometer reading parsing algorithm">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1.  Let |reading| be a new [=sensor reading=].
+    1.  Let |value| be a new [=sensor reading=].
+    1.  Let |reading| be the result of [=parse XYZ reading=] with |parameters|.
+    1.  If |reading| is "undefined".
+        1.  Return "undefined".
+    1. Let |key| be the [=/list=] « "`xBias`", "`yBias`", "`zBias`" ».
+    1.  [=map/For each=] |key| → let |value| be the result of [=parse single-value number reading=] with |parameters| and |key|.
+        1.  If |value| is "undefined".
+            1.  Return "undefined".
+        1.  [=map/Set=] |reading|[|key|] to |value|.
+    1.  Return |reading|.
```

Comment:
The comments from w3c/proximity#57 about a similar loop apply here as well.

Reply:
Fixed in https://github.com/w3c/magnetometer/pull/69/commits/feb72a14a31ffb828532784602679b4c943ca75e